sway-core/src/semantic_analysis/ast_node/expression/intrinsic_function.rs

```diff
@@ -587,7 +587,10 @@ impl ty::TyIntrinsicFunctionKind {
                     warnings,
                     errors
                 );
-                let is_valid_arg_ty = matches!(arg_ty, TypeInfo::UnsignedInteger(_));
+                let is_valid_arg_ty = matches!(
+                    arg_ty,
+                    TypeInfo::UnsignedInteger(_) | TypeInfo::RawUntypedPtr
```

Comment:
Does this mean that I can now do `__add(raw_ptr_1, raw_ptr_2)`? What about `__add(5, raw_ptr_2)`? I think both should be disallowed.

Reply:
RHS is checked below and shouldn't allow that. If LHS is rq_ptr, RHS must be u64.

Reply:
Makes sense. I guess the type checker will complain if `with_type_annotation` contains the wrong type.

One more thing: the current code allows `__mul` and `__div` which I assume we don't want to allow for `raw_ptr`. Maybe a small check somewhere could be useful?